controllers/complianceeventsapi/server.go

```diff
@@ -446,9 +503,145 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
+		unAuthorizedClusters, includeAllManagedClusters, err := getAuthorizedClusters(ctx,
+		db, *parsed, userConfig, client)
+	if err != nil {
+		return nil, err
+	}
+
+	if includeAllManagedClusters {
+		// parsed.Filters["clusters.name"] and parsed.Filters["clusters.cluster_id"] are empty
+		return parsed, nil
+	}
+
+	// If there is any unAuthorized Cluster, it should return ErrNotAuthorized
+	if len(unAuthorizedClusters) > 0 {
+		log.V(2).Info("need GET access", "managedClusters", strings.Join(unAuthorizedClusters, ", "))
+
+		return nil, fmt.Errorf("need GET access to these managedClusters: %s, error: %w",
+			strings.Join(unAuthorizedClusters, ", "), ErrNotAuthorized)
+	}
+
+	if len(authorizedClusterNames) == 0 && len(authorizedClusterIds) == 0 {
+		log.V(2).Info("The service account does not have GET authorization for all")
+
+		return nil, ErrNotAuthorized
+	}
+
+	parsed.Filters["clusters.name"] = authorizedClusterNames
+	parsed.Filters["clusters.cluster_id"] = authorizedClusterIds
+
 	return parsed, nil
 }
 
+// Return authorized cluster names or, If the url query doesn't have specific clusters, it verifies
+// all managedCluster Authorizations and it returns true if the user has all authorization for manageClusters
+// getAuthorizedClusterNames returns also unauthorized managedClusters name list to give information to users
+func getAuthorizedClusters(ctx context.Context, db *sql.DB, parsed queryOptions,
+	userConfig *rest.Config, c client.Client,
+) (authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters []string,
+	includeAllManagedClusters bool, err error,
+) {
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters, false, err
+	}
+
+	clusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	if okClusterIds {
+		// Convert id to name
+		for _, id := range clusterIDs {
+			clusterName, err := getClusterNameFromID(ctx, db, id)
+			if err != nil {
+				log.Error(err, "Failed to get cluster name from cluster id :"+id)
+
+				// This leads to send forbidden error to the user
+				unAuthorizedClusters = append(unAuthorizedClusters, id)
+
+				continue
+			}
+
+			if getAccessByClusterName(allRules, clusterName) {
+				authorizedClusterIds = append(authorizedClusterIds, id)
+			} else {
+				unAuthorizedClusters = append(unAuthorizedClusters, id)
+			}
+		}
+	}
+
+	parsedClusterNames, okClusterNames := parsed.Filters["clusters.name"]
+	if okClusterNames {
+		for _, clusterName := range parsedClusterNames {
+			if getAccessByClusterName(allRules, clusterName) {
+				authorizedClusterNames = append(authorizedClusterNames, clusterName)
+			} else {
+				unAuthorizedClusters = append(unAuthorizedClusters, clusterName)
+			}
+		}
+	}
+
+	// There is no cluster.id or clusterName arg from the url query
+	// In other words, the user requests all
+	if !okClusterIds && !okClusterNames {
```

Comment:
Optional: instead of this indent, you could just have this:

```go
if okClusterIds || okClusterNames {
return authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters, false, nil
}
```

Reply:
I am using `return authorizedClusterIds, authorizedClusterNames, unAuthorizedClusters, false, nil`, result of this block so